curl/src/main/scala/org/http4s/curl/unsafe/libcurl.scala

```diff
@@ -100,8 +100,12 @@ private[curl] object libcurl {
 
   type CURLoption = CUnsignedInt
 
+  type CURLversion
```

Comment:
Hm, does this not need a more precise type? e.g. ` = CUnsignedInt` I'm kind of surprised its working (is it just treating it as a 64-bit pointer)?

Reply:
Actually it's the name of the base enum type that represents the version of struct that contains the version, and is not an integer.

Reply:
Sure, but it must be some concrete type i.e. what is its size? :)

Reply:
I thought it is treated as external, like `CURL` or `CURLM` for example, isn't that the case here?

Reply:
> like `CURL` or `CURLM` for example

Do we ever use those raw types directly? I believe they are always used like `Ptr[CURL]` or `Ptr[CURLM]` which is fine, because the size of the pointer is known (and it's not necessary to know the size of what it's pointing to, to pass the pointer around).

Reply:
🤔 Maybe it's working by mere chance, or because enum has a size that fits any matched size (e.g. an octet).
Does adding an explict type solve the problem? Or is there a way to test it?

Reply:
> Maybe it's working by mere chance

Yes, I suspect that's the case. In my experience (if I remember correctly) enums are typically encoded as 32-bit ints. My guess is that SN is treating this as an instance of a Scala object, which are passed with (64-bit) pointers. So it seems to be working okay as its currently used.

---

As an experiment, you can try defining a method like this in your C glue code.
```c
int curlversion_test(x: CURLversion, y: CURLversion, z: CURLversion) {
  if (x == y && y == z) return 1;
  else return 0;
}
```
and then calling from Scala like this:
```scala
val v = org_http4s_curl_version_now()
assert(curlversion_test(v, v, v) == 1)
```

I have a suspicion this might go wrong, if enums are represented by 32-bit ints.

---

> Does adding an explict type solve the problem?

Yes, this is what we should do :)